packages/astro/src/env/env-loader.ts

```diff
@@ -22,45 +25,51 @@ function getPrivateEnv(
 	const privateEnv: Record<string, string> = {};
 	for (const key in fullEnv) {
 		// Ignore public env var
-		if (isValidIdentifierRe.test(key) && envPrefixes.every((prefix) => !key.startsWith(prefix))) {
-			if (typeof process.env[key] !== 'undefined') {
-				let value = process.env[key];
-				// Replacements are always strings, so try to convert to strings here first
-				if (typeof value !== 'string') {
-					value = `${value}`;
-				}
-				// Boolean values should be inlined to support `export const prerender`
-				// We already know that these are NOT sensitive values, so inlining is safe
-				if (
-					!useRawValues &&
-					(value === '0' || value === '1' || value === 'true' || value === 'false')
-				) {
-					privateEnv[key] = value;
-				} else {
-					privateEnv[key] = `process.env.${key}`;
-				}
+		if (!isValidIdentifierRe.test(key) || envPrefixes.some((prefix) => key.startsWith(prefix))) {
+			continue;
+		}
+		// Only replace with process.env if not static
+		// TODO: make static the default and only way to do this in Astro 6
+		if (!useStatic && typeof process.env[key] !== 'undefined') {
+			let value = process.env[key];
+			// Replacements are always strings, so try to convert to strings here first
+			if (typeof value !== 'string') {
+				value = `${value}`;
+			}
+			// Boolean values should be inlined to support `export const prerender`
+			// We already know that these are NOT sensitive values, so inlining is safe
+			if (value === '0' || value === '1' || value === 'true' || value === 'false') {
+				privateEnv[key] = value;
```

Comment:
This could be quite confusing if boolean values are the only ones that use build-time values while the others use runtime

Reply:
FYI this is just the previous behavior so I don't think it's more confusing than before